### DISCUSSION TITLE
Load svg as a string

### DISCUSSION THREAD

**Discussion by devdivr** [coil-kt/coil#1031] [Q&A]
Hi, 
I found ways to load svg from a file, urls and etc but think coil doesn't support to load svg from a string like below.
```
<svg xmlns="http://www.w3.org/2000/svg" width="2048" height="1024">  <g><defs>   <linearGradient id="gradientae3b9474-b95d-4b73-8262-bde6130ab5eb" x1="0%" y1="0%" x2="0%" y2="100%">     <stop offset="0%" stop-color="black" stop-opacity="0" />     <stop offset="70%" stop-color="white" stop-opacity="1" />     <stop offset="100%" stop-color="white" stop-opacity="1" />   </linearGradient>   <mask id="gradient-maskae3b9474-b95d-4b73-8262-bde6130ab5eb">     <rect x="794.2145775747783" y="296.6421041159125" width="61.44" height="61.44" fill="url(#gradientae3b9474-b95d-4b73-8262-bde6130ab5eb)"  />   </mask>  </defs>    <circle      cx="824.9345775747784"      cy="327.36210411591253"      stroke="red"      stroke-width="10.239999999999998"      r="25.6"      fill="transparent"      mask="url(#gradient-maskae3b9474-b95d-4b73-8262-bde6130ab5eb)"    /></g><g>                    <mask id="arrowTip-mask5b8f704b-851e-446f-ac3f-715f65c77656" maskUnits="userSpaceOnUse" x="0" y="0" width="2048" height="1024">                        <rect x="0" y="0" width="2048" height="1024" fill="rgb(255,255,255)"/>                        <rect x="665.7925453901003" y="781.3897022914837" width="22.518936140931636" height="22.518936140931636" fill="rgb(0,0,0)" transform="rotate(197.18635668513696, 677.0520134605661, 781.3897022914837)"/>                    </mask>  <path d="M 817.3637159328711 327.73344515584677 Q 747.2078646967187 554.5615737236651, 677.0520134605661 781.3897022914837" stroke-dasharray="10.24 0" stroke="red" stroke-width="10.24" fill="transparent" mask="url(#arrowTip-mask5b8f704b-851e-446f-ac3f-715f65c77656)"/>  <polygon points="677.0520134605661,781.3897022914837 665.7925453901003, 803.9086384324153 688.3114815310319,803.9086384324153" transform="rotate(197.18635668513696, 677.0520134605661, 781.3897022914837)" fill="red"/></g></svg>
```

One solution to load a string I'm able to try is to save string as a file then load it again but it's not efficient. 
Can anyone know how to load svg string?

**Comment by colinrtwhite** [comment]
Coil 2.x supports loading `ByteBuffer`s so I think you could convert the string to a byte buffer then load it: `imageView.load(ByteBuffer.wrap("<svg xmlns="http://www.w3.org/2000/svg...".toByteArray()))`.

**Reply by akhukhra** [reply]
How can i achieve same result in jetpack compose?

**Comment by akhukhra** [comment]
> How can i achieve same result in jetpack compose?


upd:
use 
	  implementation "io.coil-kt:coil-compose:2.4.0"
	  implementation "io.coil-kt:coil-compose-base:2.4.0"
	  implementation "io.coil-kt:coil-svg:2.4.0"

and this

        val model = ImageRequest.Builder(context = LocalContext.current)
		        .data(ByteBuffer.wrap(svg.toByteArray()))
		        .decoderFactory(SvgDecoder.Factory())
		        .decoderDispatcher(Dispatchers.IO)
		        .build()
	        
	val imageLoader = LocalContext.current.imageLoader.newBuilder()
		.logger(DebugLogger())
		.build()
	
	AsyncImage(
		model = model, imageLoader = imageLoader, contentDescription = null, modifier = Modifier.size(256.dp)
	)
